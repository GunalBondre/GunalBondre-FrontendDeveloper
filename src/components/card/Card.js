import React from 'react';
import PropTypes from 'prop-types';

import { rocketImg } from '../../assets';

const Card = ({ item }) => {
	return (
		<div
			className='max-w-md rounded overflow-hidden shadow-lg'
			key={item.capsule_serial}
			data-testid='card'
		>
			<img className='w-full' src={rocketImg} alt='rocket' />
			<div className='px-6 py-4'>
				<div className='font-bold text-xl mb-2'>status - {item.status}</div>
				<p className='text-gray-700 text-base'>details - {item.details}</p>
			</div>
			<div className='px-6 pt-4 pb-2'>
				<span className='inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2'>
					type - {item.type}
				</span>
				<span className='inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2'>
					launch - {item.original_launch}
				</span>
				<span className='inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2'>
					landing - {item.landings}
				</span>
			</div>
		</div>
	);
};
Card.propTypes = {
	item: PropTypes.shape({
		type: PropTypes.string,
		original_launch: PropTypes.string,
		landings: PropTypes.number,
		status: PropTypes.string,
		details: PropTypes.string,
		capsule_serial: PropTypes.string,
	}),
};
export default Card;
