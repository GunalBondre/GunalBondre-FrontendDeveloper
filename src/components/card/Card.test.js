import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Card from './Card';

const mockData = {
	status: 'active',
	type: 'Dragon 1.0',
	details: 'lorem i[sum doler',
	original_launch: '2 aug 2019',
	landings: 1,
};

describe('Card Component', () => {
	it('should render the component', () => {
		render(<Card item={mockData} />);

		expect(screen.getByTestId('card')).toBeDefined();
	});

	it('should match the snapshot', () => {
		expect(render(<Card item={mockData} />)).toMatchSnapshot();
	});
});
