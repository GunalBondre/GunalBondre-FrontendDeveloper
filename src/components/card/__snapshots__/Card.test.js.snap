// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-md rounded overflow-hidden shadow-lg"
        data-testid="card"
      >
        <img
          alt="rocket"
          class="w-full"
          src="rocket1.jpg"
        />
        <div
          class="px-6 py-4"
        >
          <div
            class="font-bold text-xl mb-2"
          >
            status - 
            active
          </div>
          <p
            class="text-gray-700 text-base"
          >
            details - 
            lorem i[sum doler
          </p>
        </div>
        <div
          class="px-6 pt-4 pb-2"
        >
          <span
            class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
          >
            type - 
            Dragon 1.0
          </span>
          <span
            class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
          >
            launch - 
            2 aug 2019
          </span>
          <span
            class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
          >
            landing - 
            1
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-md rounded overflow-hidden shadow-lg"
      data-testid="card"
    >
      <img
        alt="rocket"
        class="w-full"
        src="rocket1.jpg"
      />
      <div
        class="px-6 py-4"
      >
        <div
          class="font-bold text-xl mb-2"
        >
          status - 
          active
        </div>
        <p
          class="text-gray-700 text-base"
        >
          details - 
          lorem i[sum doler
        </p>
      </div>
      <div
        class="px-6 pt-4 pb-2"
      >
        <span
          class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
        >
          type - 
          Dragon 1.0
        </span>
        <span
          class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
        >
          launch - 
          2 aug 2019
        </span>
        <span
          class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
        >
          landing - 
          1
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
